back/src/Guide/Database/Add.hs

```diff
@@ -0,0 +1,258 @@
+  , itemRowSummary     :: Text
+  , itemRowEcosystem   :: Text
+  , itemRowNotes       :: Text
+  , itemRowDeleted     :: Bool
+  , itemRowCategoryUid :: Uid Category
+  , itemRowProsOrder   :: [Uid Trait]
+  , itemRowConsOrder   :: [Uid Trait]
+  }
+
+-- | Pass a 'ItemRow' to query.
+itemRowParams :: HE.Params ItemRow
+itemRowParams =
+  (itemRowUid >$< uidParam) <>
+  (itemRowName >$< textParam) <>
+  (itemRowCreated >$< timestamptzParam) <>
+  (itemRowLink >$< textParamNullable) <>
+  (itemRowHackage >$< textParamNullable) <>
+  (itemRowSummary >$< textParam) <>
+  (itemRowEcosystem >$< textParam) <>
+  (itemRowNotes >$< textParam) <>
+  (itemRowDeleted >$< boolParam) <>
+  (itemRowCategoryUid >$< uidParam) <>
+  (itemRowProsOrder >$< uidsParam) <>
+  (itemRowConsOrder >$< uidsParam)
+
+-- | Insert item to database.
+addItem
+  :: Uid Category       -- ^ Category id
+  -> Uid Item           -- ^ New item's id
+  -> "name" :! Text     -- ^ Name
+  -> UTCTime            -- ^ Creation time
+  -> ExceptT DatabaseError Transaction ()
+addItem catId itemId (arg #name -> name) created = do
+  let sql = [r|
+        INSERT INTO items
+          (uid, name, created, link, hackage, summary, ecosystem, notes, category_uid, pros_order, cons_order)
+        VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11)
+        |]
+      encoder = itemRowParams
+      decoder = HD.noResult
+  lift $ HT.statement
+    ItemRow
+      { itemRowUid = itemId
+      , itemRowName = name
+      , itemRowCreated = created
+      , itemRowLink = Nothing
+      , itemRowHackage = Nothing
+      , itemRowSummary = ""
+      , itemRowEcosystem = ""
+      , itemRowNotes = ""
+      , itemRowDeleted = False
+      , itemRowCategoryUid = catId
+      , itemRowProsOrder = []
+      , itemRowConsOrder = [] }
+    (Statement sql encoder decoder False)
+  -- Adds itemId to category's items_order list.
+  addItemIdToCategory catId itemId
+
+----------------------------------------------------------------------------
+-- addTrait
+----------------------------------------------------------------------------
+
+-- | Trait intermediary type.
+data TraitRow = TraitRow
+  { traitRowUid     :: Uid Trait
+  , traitRowContent :: Text
+  , traitRowDeleted :: Bool
+  , traitRowType    :: TraitType
+  , traitRowItemUid :: Uid Item
+  }
+
+-- | Pass a 'TraitRow' to query.
+traitRowParams :: HE.Params TraitRow
+traitRowParams =
+  (traitRowUid >$< uidParam) <>
+  (traitRowContent >$< textParam) <>
+  (traitRowDeleted >$< boolParam) <>
+  (traitRowType >$< traitTypeParam) <>
+  (traitRowItemUid >$< uidParam)
+
+-- | Insert trait to database.
+addTrait
+  :: Uid Item             -- ^ Item id
+  -> Uid Trait            -- ^ New trait's id
+  -> TraitType            -- ^ Pro or Con
+  -> "content" :! Text    -- ^ Trait content
+  -> ExceptT DatabaseError Transaction ()
+addTrait itemId traitId traitType (arg #content -> content) = do
+  let sql = [r|
+        INSERT INTO traits (uid, content, type_, item_uid)
```

Comment:
Won't work. The row contains five parameters so you should add `deleted` too.

Come to think of it, we will never use the default for `deleted` so we might just as well remove it from the schema.